DigitalLearningSolutions.Web/Controllers/TrackingSystem/CourseSetup/AdminFieldsController.cs

```diff
@@ -140,6 +146,42 @@ BulkAdminFieldAnswersViewModel model
             );
         }
 
+        [HttpGet]
+        [Route("{customisationId:int}/AdminFields/{promptNumber:int}/Remove")]
+        public IActionResult RemoveAdminField(int customisationId, int promptNumber)
+        {
+            var adminWithAnswerCount =
+                userDataService.GetDelegateCountWithAnswerForCourseAdminField(customisationId, promptNumber);
+
+            if (adminWithAnswerCount == 0)
+            {
+                return RemoveAdminFieldAndRedirect(customisationId, promptNumber);
+            }
+
+            var promptName =
+                customPromptsService.GetPromptNameForCourseAndPromptNumber(customisationId, promptNumber);
+
+            var model = new RemoveAdminFieldViewModel(customisationId, promptName, adminWithAnswerCount);
+
+            return View(model);
+        }
+
+        [HttpPost]
+        [Route("{customisationId:int}/AdminFields/{promptNumber:int}/Remove")]
+        public IActionResult RemoveAdminField(int customisationId, int promptNumber, RemoveAdminFieldViewModel model)
+        {
+            if (!model.Confirm)
+            {
+                ModelState.AddModelError(
+                    nameof(RemoveAdminFieldViewModel.Confirm),
+                    "You must confirm before deleting this prompt"

```

Comment:
This message should probably be field not prompt

Reply:
changed!